### DISCUSSION TITLE
create new pipeline for Persian

### DISCUSSION THREAD

**Discussion by aliebrahiiimi** [explosion/spaCy#8233] [Language Support]
as you know there isn't any spacy pipeline for Persian, I read the document but I can't find it.
can you help how can I create a pipeline for Persian?
can I do it by myself?

**Comment by polm** [comment]
Take a look at the master thread about adding new languages. https://github.com/explosion/spaCy/discussions/3056

Adding a new language can be a lot of work, but if there's already public datasets you may be able to do it by yourself.

**Reply by polm** [reply]
Actually, it looks like someone trained Persian models in 2018. We didn't end up hosting them due to licensing issues I think, but this thread may be a good starting point. https://github.com/explosion/spaCy/pull/2797

**Reply by aliebrahiiimi** [reply]
> Take a look at the master thread about adding new languages. #3056
> 
> Adding a new language can be a lot of work, but if there are already public datasets you may be able to do it by yourself.

I see this, but I think this tutorial is for languages that have pipelines, I don't know how to add a new pipeline, is there any tutorial on the spacy site?

**Reply by polm** [reply]
spaCy already has alpha support for Persian, which means tokenization should work, and all you have to do is add models, which is what that thread is about. 

For example, this code runs for me:

```
import spacy

nlp = spacy.blank("fa")

text = "این یک جمله آزمایشی است."

for word in nlp(text):
    print(word)
```

**Reply by aliebrahiiimi** [reply]
thanks i will check it
